src/main/java/net/neoforged/neoforge/event/tick/LivingTickEvent.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.tick;
+
+import net.minecraft.world.entity.LivingEntity;
+import net.neoforged.bus.api.ICancellableEvent;
+import net.neoforged.neoforge.event.entity.living.LivingEvent;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * Base class of the two player tick events.
+ * 
+ * @see Pre
+ * @see Post
+ */
+public abstract class LivingTickEvent extends LivingEvent {
+    @ApiStatus.Internal
+    public LivingTickEvent(LivingEntity entity) {
+        super(entity);
+    }
+
+    /**
+     * {@link LivingTickEvent.Pre} is fired once per game tick, per entity, before the entity performs work for the current tick.
+     * <p>
+     * This event fires on both the logical server and logical client.
+     */
+    public static class Pre extends LivingTickEvent implements ICancellableEvent {
+        public Pre(LivingEntity entity) {
+            super(entity);
+        }
+
+        /**
+         * Cancels this event, preventing the current tick from being executed for the entity.
```

Comment:
This is not fully true for all entities. There are plenty of entity classes that run code in `tick` before calling super which is not cancelled.

The ones I've found in vanilla are: Breeze, Creeper, Endermite, Frog, Pufferfish, ShoulderRidingEntity, Silverfish, Skeleton, Slime, Sniffer, Strider, Vex, Warden, Zombie, ZombieVillager